lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -680,65 +680,97 @@ sub done {
     $self->render(json => {result => \$res, reason => \$reason});
 }
 
-=over 4
-
-=item restart()
-
-Restart job(s).
-
-Use force=1 to force the restart (e.g. despite missing assets).
-Use skip_parents=1 to prevent restarting parent jobs.
-Use skip_children=1 to prevent restarting child jobs.
-Use skip_ok_result_children=1 to prevent restarting passed/softfailed child jobs.
-
-Used for both apiv1_restart and apiv1_restart_jobs
-
-=back
-
-=cut
-
-sub restart {
-    my ($self) = @_;
+sub _restart {
+    my ($self, %args) = @_;
 
-    my @flags      = qw(force skip_parents skip_children skip_ok_result_children);
+    my $dup_route  = $args{duplicate_route_compatibility};
```

Comment:
I would prefer setting the real args here i.e. `skip_aborting_jobs`, `force` rather than special-casing this

Reply:
But this is just about the defaults - the "duplicate" route now supports all parameters the "restart" route does (and vice versa). I don't want to duplicate code for the argument validation so I thought it is best to pass a single flag here. Also note that this argument slightly changes the format of the resulting array.